packages/services/api/src/shared/entities.ts

```diff
@@ -54,9 +54,27 @@ export interface PersistedOperation {
 
 export const emptySource = '*';
 
+export class GraphQLDocumentStringInvalidError extends Error {
+  constructor(message: string, location?: SourceLocation) {
+    const locationString = location ? ` at line ${location.line}, column ${location.column}` : ''
+    super(`The provided SDL is not valid${locationString}\n: ${message}`);
+  }
+}
+
 export function createSchemaObject(schema: Schema): SchemaObject {
+  let document: DocumentNode;
+  
+  try {
+    document = parse(schema.source)
+  }  catch (err) {
+    if (err instanceof GraphQLError) {
+      throw new GraphQLDocumentStringInvalidError(err.message,  err.locations?.[0])
```

Comment:
To be honest we could also use `HiveError` here. But I did not check all other usages of `createSchemaObject` and whether those are safe to be exposed to the outside world.